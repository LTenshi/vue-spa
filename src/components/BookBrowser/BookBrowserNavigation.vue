<template>
    <th class="container" colspan="4">
        <button @click="moveBack" :style="isBackPossible" :disabled="isBackPossible">&#10094;</button>
        Page {{currPage}} of {{maxPages}} 
        <button @click="moveForward" :style="isForwardPossible" :disabled="isForwardPossible">&#10095;</button>
    </th>
</template>

<script>
export default {
 props: {
    maxPages: {
      type: Number,
      required: true
    },
    currPage: {
        type: Number,
        required: true,
    }
  },
  methods: {
    moveBack() {
        this.$emit("navigationEvent", this.currPage - 1);
    },
    moveForward() {
        this.$emit("navigationEvent", this.currPage + 1);
    }
  },
  computed: {
    isBackPossible() {
        return this.currPage === 1 ? { color: "lightgrey", cursor: "unset" } : false
    },
    isForwardPossible() {
        return this.currPage === this.maxPages ? { color: "lightgrey", cursor: "unset" } : false
    }
  }
}
</script>

<style>
.container {
    width: 100%;
}

button {
	background: none;
    color: grey;
	border: none;
	font: inherit;
	cursor: pointer;
}
</style>