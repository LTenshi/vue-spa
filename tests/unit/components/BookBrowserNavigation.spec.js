import BookBrowserNavigation from "../../../src/components/BookBrowser/BookBrowserNavigation";
import { mount } from "@vue/test-utils";

describe("Book Browser Navigation", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(BookBrowserNavigation, {
            props: {
                maxPages: 3,
                currPage: 3,
            }
        });
    })

    it("Should match the snapshot", () => {
        expect(wrapper.element).toMatchSnapshot();
    })

    it('Should not allow forward movement', () => {
        expect(wrapper.vm.isForwardPossible).toStrictEqual({"color": "lightgrey", "cursor": "unset"});
    });

    it('Should allow back movement', () => {
        //in this case false means that item is not disabled
        expect(wrapper.vm.isBackPossible).toBe(false);
    });

    it('Should display page 3 of 3', () => {
        expect(wrapper.html()).toContain("Page 3 of 3");
    });
    
});