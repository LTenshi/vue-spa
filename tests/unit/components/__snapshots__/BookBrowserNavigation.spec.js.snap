// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Browser Navigation Should match the snapshot 1`] = `
<th
  class="container"
  colspan="4"
>
  <button>
    ❮
  </button>
   Page 3 of 3 
  <button
    disabled=""
    style="color: lightgrey; cursor: unset;"
  >
    ❯
  </button>
</th>
`;
